cpp/src/arrow/status.h

```diff
@@ -302,6 +302,7 @@ class ARROW_EXPORT Status : public util::EqualityComparable<Status>,
   /// \brief Return a string representation of the status code, without the message
   /// text or POSIX code information.
   std::string CodeAsString() const;
+  static std::string CodeAsString(StatusCode);
```

Comment:
Yes. Previously `Status(expected_code, "").CodeAsString()` was used, which fails when the expected code is OK because we assert that `Status`es are never constructed with a non error code.